packages/lib/gpcircuits/circuits/proto-pod-gpc.circom

```diff
@@ -108,9 +109,61 @@ template ProtoPODGPC (
         entryValueHashes[i] <== entryProofSiblings[i][0];
     }
 
-    // Entry can be compared for equality (by hash) to another entry (by index).
-    // This can be disabled by comparing to self: entryEqualToOtherEntryIndex[i] = i
-    signal input entryEqualToOtherEntryByIndex[MAX_ENTRIES];
+    /*
+     * Virtual entries. These are entries derived from PODs'
+     * cryptographic data, e.g. the signers' public keys.
+     *
+     * This block pipes them in and forms signals to be fed into the
+     * entry constraint, tuple and list membership modules below, as well
+     * as those revealing their hashes where applicable.
+     */
+    
+    // Maximum number of virtual entries.
```

Comment:
This block might benefit from being split out into a separate template/module.  It might not be that much shorter, but it would clearly separate which parts are argument routing vs. which parts are the  logic of what a virtual entry module does.